compiler/src/dotty/tools/dotc/transform/ElimStaticThis.scala

```diff
@@ -27,10 +27,10 @@ class ElimStaticThis extends MiniPhaseTransform {
   override def transformIdent(tree: tpd.Ident)(implicit ctx: Context, info: TransformerInfo): tpd.Tree = {
     if (ctx.owner.enclosingMethod.is(JavaStatic)) {
       tree.tpe match {
-        case TermRef(thiz: ThisType, _) if thiz.cls.is(ModuleClass) =>
+        case TermRef(thiz: ThisType, _) if thiz.cls.is(ModuleClass, JavaDefined) =>
```

Comment:
I'm also not sure why you check JavaDefined if this is for packages, why not check `Package` ?